stuffer/s2n_stuffer_text.c

```diff
@@ -65,10 +65,13 @@ int s2n_stuffer_skip_whitespace(struct s2n_stuffer *s2n_stuffer)
 
 int s2n_stuffer_read_expected_str(struct s2n_stuffer *stuffer, const char *expected)
 {
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(stuffer));
+    notnull_check(expected);
     void *actual = s2n_stuffer_raw_read(stuffer, strlen(expected));
```

Comment:
If this fails, we will still have advanced the stuffer by sizeof(actual). Is that what we indent? @camshaft @mattsb42-aws @alexw91

Reply:
Probably not, i would say. We'd need a way to peek a certain number of bytes and then advance the cursor after all of the checks have passed, including the memcmp.

Reply:
I updated the implementation of `s2n_stuffer_read_expected_str` to not use `s2n_stuffer_raw_read` anymore, so we can guarantee that `read_cursor` won’t change in case of failure. Any suggestions? @danielsn @camshaft